### PR TITLE
BPMSPL-252: [GSS-RFE] Ability to check duplicate of ReleaseId (GroupId, ArtifactId and Version) with existing projects

### PR DESCRIPTION
**Pull request by manstis**
See https://issues.jboss.org/browse/BPMSPL-252

### CODE REVIEW

**Comment by manstis** [comment]
Part of the following ensemble:

https://github.com/droolsjbpm/drools/pull/633
https://github.com/droolsjbpm/guvnor/pull/240
https://github.com/droolsjbpm/kie-wb-common/pull/236
https://github.com/droolsjbpm/drools-wb/pull/117
https://github.com/droolsjbpm/kie-wb-distributions/pull/187

**Comment by manstis** [comment]
@Rikkola Would you mind taking a look please?

**Comment by manstis** [comment]
*Documentation*
https://github.com/droolsjbpm/kie-docs/pull/40

File: kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-backend/src/main/java/org/kie/workbench/common/screens/projecteditor/backend/server/ProjectScreenServiceImpl.java [hunk 1]
```diff
@@ -15,34 +15,45 @@
...
-    private ProjectScreenModelSaver  saver;
+    private ProjectScreenModelSaver saver;
 
     public ProjectScreenServiceImpl() {
+        //WELD proxy
     }
 
     @Inject
     public ProjectScreenServiceImpl( final KieProjectService projectService,
+                                     final ProjectRepositoryResolver<KieProject> repositoryResolver,
```
Comment (Rikkola) [inline kie-wb-common-screens/kie-wb-common-project-editor/kie-wb-common-project-editor-backend/src/main/java/org/kie/workbench/common/screens/projecteditor/backend/server/ProjectScreenServiceImpl.java]: Should this use KieRepositoryResolver? 
I remember some containers had trouble injecting with generics. I think the issue back then was with ProjectService<KieProject>.

File: kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/builder/BuildServiceImpl.java [hunk 2]
```diff
@@ -71,11 +79,13 @@ public BuildServiceImpl() {
     public BuildServiceImpl( final POMService pomService,
                              final ExtendedM2RepoService m2RepoService,
                              final KieProjectService projectService,
+                             final ProjectRepositoryResolver<KieProject> repositoryResolver,
```
Comment (Rikkola) [inline kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/builder/BuildServiceImpl.java]: Same thing as above. KieRepositoryResolver would be the safe way.

File: kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/project/KieProjectServiceImpl.java [hunk 3]
```diff
@@ -69,7 +73,8 @@ public KieProjectServiceImpl( final @Named( "ioStrategy" ) IOService ioService,
                                   final AuthorizationManager authorizationManager,
                                   final BackwardCompatibleUtil backward,
                                   final CommentedOptionFactory commentedOptionFactory,
-                                  final KieResourceResolver resourceResolver ) {
+                                  final KieResourceResolver resourceResolver,
+                                  final ProjectRepositoryResolver<KieProject> repositoryResolver ) {
```
Comment (Rikkola) [inline kie-wb-common-services/kie-wb-common-services-backend/src/main/java/org/kie/workbench/common/services/backend/project/KieProjectServiceImpl.java]: And here, just highlighting what I find.

**Comment by Rikkola** [comment]
Only the ProjectRepositoryResolver issue popped up. If my memory is wrong and generics are safe to use when injecting then +1. I think the issue was when injecting with generics in WebLogic or WebSphere.

**Comment by manstis** [comment]
@Rikkola Use of ```ProjectRepositoryResolver<KieProject>``` has been replaced with ```KieRepositoryResolver```

**Comment by Rikkola** [comment]
+1

**Comment by csadilek** [comment]
master: 627aff35ce61cacd92215b9f9d2c0eb860a7edb1

**Pull request closed without merging**